pkg/tc/server.go

```diff
@@ -0,0 +1,150 @@
+	"github.com/codenotary/immudb/pkg/logger"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+	"time"
+
+	immuclient "github.com/codenotary/immudb/pkg/client"
+	"github.com/codenotary/immudb/pkg/server"
+)
+
+// NewServer return a new trust checker server
+func NewServer(options Options) (immutcServer *ImmuTcServer, err error) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	immutcServer = DefaultServer()
+
+	immutcServer.Options = options
+
+	cliOpts := &immuclient.Options{
+		Address:            options.ImmudbAddress,
+		Port:               options.ImmudbPort,
+		HealthCheckRetries: 1,
+		MTLs:               options.MTLs,
+		MTLsOptions:        options.MTLsOptions,
+		Auth:               false,
+		Config:             "",
+	}
+
+	if options.Logfile != "" {
+		var flogger logger.Logger
+		if flogger, immutcServer.LogFile, err = logger.NewFileLogger("immutc ", options.Logfile); err == nil {
+			immutcServer.WithLogger(flogger)
+		} else {
+			return nil, err
+		}
+	}
+	var ic immuclient.ImmuClient
+	if ic, err = immuclient.NewImmuClient(cliOpts); err != nil {
+		immutcServer.Logger.Errorf("Unable to instantiate client %s", err)
+		return nil, err
+	}
+
+	immutcServer.WithChecker(NewImmuTc(ic, immutcServer.Logger))
+
+	return immutcServer, nil
+}
+
+// Start launch trust checker and status http server
+func (s *ImmuTcServer) Start() (err error) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	defer s.LogFile.Close()
+	s.Logger.Infof("starting immutc: %v", s.Options)
+	if s.Options.Pidfile != "" {
+		if s.Pid, err = server.NewPid(s.Options.Pidfile); err != nil {
+			s.Logger.Errorf("failed to write pidfile: %s", err)
+			return err
+		}
+	}
+
+	go func() error {
+		if err = s.Checker.Start(ctx); err != nil {
+			return err
+		}
+		return nil
+	}()
+
+	s.HttpServer = s.newWebserver()
+	go s.gracefulShutdown()
+
+	s.Logger.Infof("Status server is ready to handle requests at %s/status", s.Options.Address+":"+strconv.Itoa(s.Options.Port))
+	if err = s.HttpServer.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+		s.Logger.Errorf("Could not listen on %s: %v", s.Options.Address+":"+strconv.Itoa(s.Options.Port), err)
+	}
+
+	<-s.Done
+	s.Logger.Infof("Trust checker stopped")
+
+	return err
+}
+
+// Stop shutdown trust checker and status http server
+func (s *ImmuTcServer) Stop() error {
+	s.gracefulShutdown()
+	return nil
+}
+
+func (s *ImmuTcServer) newWebserver() *http.Server {
+	router := http.NewServeMux()
+	router.HandleFunc("/status", func(w http.ResponseWriter, r *http.Request) {
+
+		w.Header().Set("Content-Type", "application/json")
+		status := make(map[string]bool)
+		status["status"] = s.Checker.GetStatus(r.Context())
+		if err := json.NewEncoder(w).Encode(status); err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+		return
+	})
+
+	return &http.Server{
+		Addr:         s.Options.Address + ":" + strconv.Itoa(s.Options.Port),
```

Comment:
Can be rewritten with fmt.Sprintf instead of converting string and manually parsing .